glide-core/src/socket_listener.rs

```diff
@@ -311,6 +312,44 @@ async fn send_command(
         .map_err(|err| err.into())
 }
 
+// Parse the cluster scan command parameters from protobuf and send the command to redis-rs.
+async fn cluster_scan(cluster_scan: ClusterScan, mut client: Client) -> ClientUsageResult<Value> {
+    // Since we don't send the cluster scan as a usual command, but through a special function in redis-rs library,
+    // we need to handle the command separately.
+    // Specifically, we need to handle the cursor, which is not the cursor returned from the server,
+    // but the ID of the ScanStateRC, stored in the cluster scan container.
+    // We need to get the ref from the table or create a new one if the cursor is empty.
+    let cursor: String = cluster_scan.cursor.into();
+    let cluster_scan_cursor = if cursor.is_empty() {
+        ScanStateRC::new()
+    } else {
+        get_cluster_scan_cursor(cursor)?
+    };
+
+    let match_pattern_string = cluster_scan
+        .match_pattern
+        .map(|pattern| pattern.to_string());
+    let match_pattern = match_pattern_string.as_deref();
+    let count = cluster_scan.count.map(|count| count as usize);
+
+    let object_type =
+        cluster_scan
+            .object_type
+            .and_then(|object_type| match object_type.to_string().as_str() {
+                "String" => Some(redis::ObjectType::String),
+                "List" => Some(redis::ObjectType::List),
+                "Set" => Some(redis::ObjectType::Set),
+                "ZSet" => Some(redis::ObjectType::ZSet),
+                "Hash" => Some(redis::ObjectType::Hash),
+                _ => None,
```

Comment:
That's an issue - if we pass the wrong string it would be seamlessly ignored. We should return with an error if we receive a non existing object type

Reply:
I ran into this directly in the Java port:
- enums are usually upper-case in Java.
- Was sending the enum name to Rust as-is (which works against Redis itself which allows upper-case types).
- The names were inconsistent due to casing.
- The illegal names being used were silently ignored, making tests hard to debug.
We probably want to expose the Rust names as constants for other layers to consume for consistency (same with the "finished" string).

Reply:
Will return an Error and will make it constant.